apps/microtvm/riscv/template_project/microtvm_api_server.py

```diff
@@ -0,0 +1,285 @@
+        project_model_library_format_path = project_dir / MODEL_LIBRARY_FORMAT_RELPATH
+        shutil.copy2(model_library_format_path, project_model_library_format_path)
+
+        # Extract Model Library Format tarball.into <project_dir>/model.
+        extract_path = project_dir / project_model_library_format_path.stem
+        with tarfile.TarFile(project_model_library_format_path) as tf:
+            os.makedirs(extract_path)
+            tf.extractall(path=extract_path)
+
+        # Populate CRT.
+        crt_path = project_dir / "crt"
+        os.mkdir(crt_path)
+        for item in self.CRT_COPY_ITEMS:
+            src_path = standalone_crt_dir / item
+            dst_path = crt_path / item
+            if os.path.isdir(src_path):
+                shutil.copytree(src_path, dst_path)
+            else:
+                shutil.copy2(src_path, dst_path)
+
+        # Populate CMake file
+        self._populate_cmake(
+            current_dir / f"{CMAKEFILE_FILENAME}.template",
+            project_dir / CMAKEFILE_FILENAME,
+            options.get("workspace_size_bytes", WORKSPACE_SIZE_BYTES),
+            options.get("verbose"),
+        )
+
+        # Populate crt-config.h
+        crt_config_dir = project_dir / "crt_config"
+        crt_config_dir.mkdir()
+        shutil.copy2(
+            current_dir / "crt_config" / "crt_config.h",
+            crt_config_dir / "crt_config.h",
+        )
+
+        # Populate src/
+        src_dir = project_dir / "src"
+        src_dir.mkdir()
+        shutil.copy2(
+            current_dir / "src" / "main.cc",
+            src_dir / "main.cc",
+        )
+        shutil.copy2(
+            current_dir / "src" / "platform.cc",
+            src_dir / "platform.cc",
+        )
+
+    def build(self, options):
+        cmake_args = []
+        toolchain_path = options.get("toolchain_path")
+        target = options.get("target")
+        march = options.get("march")
+        cmake_args.append(f"-DTOOLCHAIN_PATH={toolchain_path}")
+        cmake_args.append(f"-DTARGET={target}")
+        cmake_args.append(f"-DMARCH={march}")
+
+        build_dir = PROJECT_DIR / "build"
+        build_dir.mkdir()
+        subprocess.check_call(["cmake", *cmake_args, ".."], cwd=build_dir)
+        subprocess.check_call(["make"], cwd=build_dir)
+
+    def flash(self, options):
+        pass  # Flashing does nothing on host.
+
+    def _set_nonblock(self, fd):
+        flag = fcntl.fcntl(fd, fcntl.F_GETFL)
+        fcntl.fcntl(fd, fcntl.F_SETFL, flag | os.O_NONBLOCK)
+        new_flag = fcntl.fcntl(fd, fcntl.F_GETFL)
+        assert (new_flag & os.O_NONBLOCK) != 0, "Cannot set file descriptor {fd} to non-blocking"
+
+    def open_transport(self, options):
+        toolchain_path = options.get("toolchain_path")
+        target = options.get("target")
+        march = options.get("march")
+        spike_args = ["spike", f"--isa={march}", os.path.join(toolchain_path, target, "bin/pk")]
```

Comment:
The `--isa` argument of spike will not always be the same as the `-march` used with clang. I would prefer having isa exposed as project option (defaulting to use the same value as march) to overwrite it if required.

Reply:
I'll add isa as project option.